common/src/main/java/com/wynntils/services/itemfilter/type/StatProviderAndFilterPair.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter.type;
+
+import com.wynntils.models.items.WynnItem;
+import java.util.List;
+
+public record StatProviderAndFilterPair<T>(ItemStatProvider<T> statProvider, StatFilter<T> statFilter) {
+    public static StatProviderAndFilterPair<Object> fromPair(ItemStatProvider itemStatProvider, StatFilter value) {
```

Comment:
Yes, I did just drop the generic type here. It is not strictly necessary here and keeping it would be really hard at this point.